### DISCUSSION TITLE
[Hardware]ESP32 with ESP01 form factor

### DISCUSSION THREAD

**Discussion by luc-github** [luc-github/ESP3D#531] [Ideas]
There are several devices that use ESP01 to add ESP3D, as ESP3D is growing with new features the ESP01S with 1M flash become too small to support ESP3D web update and new features.
The solution to upgrade the flash chip is validated but not really easy for most of people with lmited soldering skill (like me 😳 )

If you compare ESP01 dimensions 14.3mm X 24.8mm
![image](https://user-images.githubusercontent.com/8822552/100584442-6bf7ba80-32ec-11eb-9828-faa63d972ca7.png)

with and ESP32 with 16MB flash dimensions 18mm x 25.5mm,   
we can imagine a final pcb around 20mm x 30mm, which still should fit existing space used by ESP01 IMHO
![image](https://user-images.githubusercontent.com/8822552/100584590-96e20e80-32ec-11eb-9253-b6eac45e76c5.png)

All pins could be compatible - not sure how to handle RST one actually, ad bridge GPIO 15 to GND would avoid boot garbage
![image](https://user-images.githubusercontent.com/8822552/100585149-63ec4a80-32ed-11eb-85ec-b9976354ab12.png)

About cost :
* ESP01S 1M: 1$~2$
* ESP32 16M: 2$~3$ + pcb cost (1$ ?) = 3$~4$ (?)    

Difference is not big but benefits are : more memory / more flash / more cores (x2)

I feel soldering ESP32 is easier than rework the memory for limited soldering skill people if come DIY 

One thing I am not able to evaluate is the power consumption of ESP8266 vs ESP32, but I guess there are some impact - so need to be tested.

I am not electronician, so please feel free to comment, I did not see such product yet - may be there is a reason ? 

Adfditionnaly if there is a way to allow access with some header the unused pins while keeping small form factor,  it would be a great upgrade.

**Comment by alxblog** [comment]
About unused pins, 
IMHO, FFC connector with flat cable and breakout board might be a good way to deal with it.

**Comment by vivian-ng** [comment]
Hi, maybe check out the [TinyPico](https://unexpectedmaker.com/tinypico), which is a tiny ESP32 development board. Anyway, it is definitely possible to just slap a ESP32 module to a PCB breaking out the GPIO0, RX, TX, 3.3V and GND pins if compatibility with the connector for ESP-01 module is critical. The thing is that you will still need to find a way to program the ESP32 (like the ESP-01, using a USB-UART device).

**Comment by luc-github** [comment]
@vivian-ng yes, the critical part  is the connector and size to fit existing used socket, to be a pin to pin replacement, but the ESP32 seems not having a reset pin unlike ESP8266 - I do not see it 

EDIT: Ok seems I mixed CH_PD esp8266 pin as EN/RST pin - in ESP32 EN pin = RST pin

**Reply by luc-github** [reply]
seems there is a board now https://github.com/Xinyuan-LilyGO/T-01C3

**Reply by luc-github** [reply]
Received  - flashed and tested - good 😸

**Reply by barbudor** [reply]
That's an ESP32-C3 but I don't see build environment for C3 in platformio.ini
Do you have it ?

I love the C3 as it has mostly all the features of the C32 but with the cost os the ESP8266, plus there are existing modules that are pin compatible with ESP32 modules such as ESP12F

**Reply by luc-github** [reply]
https://github.com/luc-github/ESP3D/blob/3.0/platformio.ini#L202

in v3 https://github.com/luc-github/ESP3D/tree/3.0

even better, use : https://luc-github.github.io as suggested in readme

**Reply by barbudor** [reply]
Ah, I was trying 2.1.x
My understanding is that v3.x is still pretty much beta. Am I wrong ?

Nice configurator, I'll give it a try tomorrow

**Reply by luc-github** [reply]
yes currently it is in alpha but close to be in beta you right - I am late due to another project: https://github.com/luc-github/ESP3D-TFT but work done in TFT project will also benefit to ESP3D V3 - in worst case will be implemented in V3.1 TBD
I just have a lot in my hands so things go slowly
